source/constructs/api/catalog/service.py

```diff
@@ -412,14 +425,15 @@ def sync_crawler_result(
                 glue_client.batch_delete_table(DatabaseName=glue_database_name,
                                           TablesToDelete=delete_glue_table_names)
             except Exception as err:
-                logger.info("batch delete glue tables error" + str(err))
-            logger.debug("batch create columns" + str(column_create_list))
+                logger.exception("batch delete glue tables error" + str(err))
+            if logger.isEnabledFor(logging.DEBUG):
+                logger.debug("batch create columns" + str(column_create_list))
+                logger.debug("batch update columns" + str(column_update_list))
+                logger.debug("batch create tables" + str(table_create_list))
+                logger.debug("batch update tables" + str(table_update_list))
             crud.batch_create_catalog_column_level_classification(column_create_list)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We recommend that you cache the result of `isEnabledFor()` in a local or instance variable and then use that cached value instead of calling `isEnabledFor()` multiple times. `isEnabledFor()` can be an expensive call, so using its cached return value can improve the performance of your code. This is particularly important for deeply nested loggers when the only explicit level is set high in the logger hierarchy and when you don&#39;t want to call a method in a tight loop.

[Learn more](https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor)

Similar issue at line numbers 445 and 451.